packages/nextjs/src/config/define-config.test.ts

```diff
@@ -1,27 +1,374 @@
+        delete process.env.VERCEL_ENV;
+        expect(resultConfig.multisite.useCookieResolution()).to.be.false;
+      }
+    });
+
+    it('it should be able to override cookie resolution function', () => {
+      defineConfigModule.defineConfig({
+        ...defaultConfig(),
+        multisite: { useCookieResolution: () => true },
+      });
+      const resultConfig = defineConfigCoreStub.getCalls()[0].args[0];
+      expect(resultConfig.multisite?.useCookieResolution).to.be.a('function');
+      if (resultConfig.multisite?.useCookieResolution) {
+        expect(resultConfig.multisite.useCookieResolution()).to.be.true;
+      }
+    });
+  });
+
+  describe('config.personalize', () => {
+    describe('environment variable is not set', () => {
+      it('should default to undefined', () => {
+        defineConfigModule.defineConfig(defaultConfig());
+        const resultConfig = defineConfigCoreStub.getCalls()[0].args[0];
+        expect(resultConfig.personalize?.scope).to.be.undefined;
+      });
+
+      it('should use the value from the config', () => {
+        defineConfigModule.defineConfig({
+          ...defaultConfig(),
+          personalize: { scope: 'custom-scope' },
+        });
+        const resultConfig = defineConfigCoreStub.getCalls()[0].args[0];
+        expect(resultConfig.personalize?.scope).to.equal('custom-scope');
+      });
+    });
+
+    describe('environment variable is set', () => {
+      before(() => {
+        process.env.NEXT_PUBLIC_PERSONALIZE_SCOPE = 'custom-env-scope';
+      });
+
+      after(() => {
+        delete process.env.NEXT_PUBLIC_PERSONALIZE_SCOPE;
+      });
+
+      it('should use the value from the config if present', () => {
+        defineConfigModule.defineConfig({
+          ...defaultConfig(),
+          personalize: { scope: 'custom-scope' },
+        });
+        const resultConfig = defineConfigCoreStub.getCalls()[0].args[0];
+        expect(resultConfig.personalize?.scope).to.equal('custom-scope');
+      });
+
+      it('should use the env var if config value not present', () => {
+        defineConfigModule.defineConfig(defaultConfig());
+        const resultConfig = defineConfigCoreStub.getCalls()[0].args[0];
+        expect(resultConfig.personalize?.scope).to.equal('custom-env-scope');
+      });
+    });
+  });
+
+  describe('config.disableStaticPaths', () => {
     describe('environment variable is not set', () => {
       it('should default to false', () => {
-        const config = defineConfig(defaultConfig());
-        expect(config.disableStaticPaths).to.equal(false);
+        defineConfigModule.defineConfig(defaultConfig());
+        const resultConfig = defineConfigCoreStub.getCalls()[0].args[0];
+        expect(resultConfig.disableStaticPaths).to.equal(false);
       });
 
       it('should use the value from the config', () => {
-        const config = defineConfig({
+        defineConfigModule.defineConfig({
           disableStaticPaths: false,
```

Comment:
let's set it to `true` so we will test this edge case
because, currently `false` is our default value as well